server/base/src/main/java/org/apache/accumulo/server/util/TableDiskUsage.java

```diff
@@ -212,13 +214,14 @@ public void print(String line) {
     for (String tableId : tablesReferenced) {
       for (String tableDir : nameSpacesReferenced) {
         // Find each file and add its size
-        FileStatus[] files = fs.globStatus(new Path(tableDir + "/" + tableId + "/*/*"));
-        if (files != null) {
-          for (FileStatus fileStatus : files) {
-            // Assumes that all filenames are unique
-            String name = fileStatus.getPath().getName();
-            tdu.addFileSize(name, fileStatus.getLen());
-          }
+
+        Path path = new Path(tableDir + "/" + tableId + "/*/*");
+        RemoteIterator<LocatedFileStatus> ri = fs.listStatus(path, false);
+
+        while (ri.hasNext()) {
+          FileStatus status = ri.next();
+          String name = status.getPath().getName();
+          tdu.addFileSize(name, status.getLen());
```

Comment:
I wonder if this is unnecessarily adding directories.  If so, this may have been a problem with the existing code.

Reply:
That might have been a problem with the original globbing code, but I think the recursive lookup that it does now only shows files. Could be confirmed with some experimentation.